### PR TITLE
fix: send only non zero usage

### PR DESCRIPTION
**Pull request by kappa90**
## Problem
We don't need to send reports for organizations that have no usage in the period

## Changes
Skip sending reports to the billing service if usage is zero

## How did you test this code?
Added a test

### CODE REVIEW

**Comment by neilkakkar** [comment]
Zero context, but one question: How do you judge if a report failed to send, or the org had zero usage in period without this?

**Comment by kappa90** [comment]
> Zero context, but one question: How do you judge if a report failed to send, or the org had zero usage in period without this?

That's a really good point. The goal for this PR is to reduce the burden on the billing service. We're still capturing exceptions and sending events to Sentry to report if something is off, hopefully that should tell us if something is going wrong.

Additionally, I was thinking we could add something on the billing service where, if there is a gap between two reports with usage > 0, we would tell PostHog to send those missing reports to fill the gaps, as we could expect the missing day to have usage != 0. We could also do the same for organizations which suddenly stop sending events.

**Comment by neilkakkar** [comment]
Sounds good, I'm happy you've thought about it 👍

File: posthog/tasks/usage_report.py [hunk 1]
```diff
@@ -401,6 +401,11 @@ def capture_report(
     pha_client.flush()
 
 
+# extend this with future usage based products
+def has_non_zero_usage(report: FullUsageReport) -> bool:
+    return report.event_count_in_period > 0 or report.recording_count_in_period > 0
```
[the remark this answers is not in the document]
Comment (kappa90) [reply posthog/tasks/usage_report.py:406]: In the billing service, we have a `PRODUCT_USAGE_MAP`, which maps which key refers to usage for which product. We could do the same here.

**Pull request merged**